newIDE/app/src/MainFrame/index.js

```diff
@@ -1987,7 +1987,12 @@ const MainFrame = (props: Props) => {
     else if (!!fileMetadata) state = await openFromFileMetadata(fileMetadata);
 
     if (state) {
-      if (state.currentProject) state.currentProject.resetProjectUuid();
+      if (state.currentProject) {
+        const { currentProject } = state;
+        currentProject.resetProjectUuid();
+        if (fileMetadata && fileMetadata.projectName)
+          currentProject.setName(fileMetadata.projectName);
+      }
```

Comment:
As discussed, can we save the project into the json here? to avoid having wrong uuid & name in the json!

Reply:
Might be more complex than just calling a save function, but we should think about it for either this PR or another PR.

Reply:
I tried and then I realized it wouldn't be this easy indeed, I would say another PR

Reply:
Correct it's not easy because we need to act differently than a traditional save (like if it fails, we probably ignore the problem) (we also don't want to trigger the window to select a save destination like done in the web-app, so we need a flag to tell "please save silently and if it fails or there is no obvious place where to save, just don't care.")

Reply:
There's also a challenge on synchronicity: the saveProject method returns if currentProject is not defined, but at the moment of this function, it is never defined.